qubes/vm/mix/net.py

```diff
@@ -61,6 +64,27 @@ def _setter_ip(self, prop, value):
         raise ValueError('Invalid IP address value')
     return value
 
+def _default_ip6(self):
+    if not self.is_networked():
+        return None
+    if not self.features.check_with_netvm('ipv6', False):
+        return None
+    if self.netvm is not None:
+        return self.netvm.get_ip6_for_vm(self)  # pylint: disable=no-member
+
+    return self.get_ip6_for_vm(self)
+
+
+def _setter_ip6(self, prop, value):
+    # pylint: disable=unused-argument
+    if not isinstance(value, str):
+        raise ValueError('IPv6 address must be a string')
+    value = value.lower()
+    try:
+        socket.inet_pton(socket.AF_INET6, value)
+    except socket.error:
+        raise ValueError('Invalid IPv6 address value')
+    return value
```

Comment:
May I suggest using `type=ipaddress.IPv6Address` instead of this whole setter? It has validation built-in, and str(address) works well for serialising.

Reply:
Should I do the same for `ip` property? AFAIR there was some reason why I didn't wanted to use it before, but cannot find one right now. Note that this code is may be exposed to untrusted input, but I assume input validation in `ipaddress` module is safe enough.

Reply:
Replaced for both, check now.

Reply:
That was a question of compatibility with older Python versions, as `ipaddress` is a relatively recent addition to stdlib (it was a 3rd party package before then). We thought about it before qubes.

Reply:
Ah, it was probably before we've switched to python3. So, that point is mute now.